src/IMenuItem.php

```diff
@@ -32,20 +32,20 @@ public function getRealLink(): string;
 	public function getRealAbsoluteLink(): string;
 
 	/**
-	 * @return array<string, string>
+	 * @return array<string, mixed>
 	 */
 	public function getData(): array;
 
 	/**
-	 * @param array<string, string> $data
+	 * @param array<string, string|int|float|bool|array|null> $data
 	 */
 	public function setData(array $data): void;
 
 	public function hasDataItem(string $name): bool;
 
-	public function getDataItem(string $name, ?string $default = null): ?string;
+	public function getDataItem(string $name, string|int|float|bool|array|null $default = null): string|int|float|bool|array|null;
```

Comment:
Why not mixed?

Reply:
It is possible and all in all I'm not against it. I was just wondering if it is OK to let pass services via data.

Reply:
So update code to use mixed instead please.